File: backend/src/main/java/com/optiplant/inventario/producto/entity/Producto.java
package com.optiplant.inventario.producto.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Producto del catálogo común de la red (SUP-03). El nivel de stock por sucursal
 * vive en {@code inventario_sucursal}, no aquí.
 */
@Entity
@Table(name = "producto")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 60)
    private String sku;

    @Column(nullable = false, length = 160)
    private String nombre;

    @Column(name = "unidad_medida_base", nullable = false, length = 30)
    private String unidadMedidaBase;
}

File: frontend/src/features/productos/types/producto.ts
export interface Producto {
  id: number;
  sku: string;
  nombre: string;
  unidadMedidaBase: string;
}

export interface ProductoInput {
  sku: string;
  nombre: string;
  unidadMedidaBase: string;
}

export interface ProductoUpdateInput {
  nombre: string;
  unidadMedidaBase: string;
}
